arangod/Cluster/HeartbeatThread.cpp

```diff
@@ -445,15 +446,34 @@ void HeartbeatThread::runSingleServer() {
     }
     start = TRI_microtime();
     
+    if (isStopping()) {
+      break;
+    }
+
     try {
       // send our state to the agency.
       // we don't care if this fails
-      sendState();
-      if (isStopping()) {
-        break;
+      double const timeout = 1.0;
+
+      // check current local version of database objects version, and bump
+      // the global version number in the agency in case it changed. this
+      // informs other listeners about our local DDL changes
+      uint64_t currentVersion = DatabaseFeature::DATABASE->versionTracker()->current();
+      
+      if (currentVersion != lastSentVersion) {
+        AgencyOperation incrementVersion("Plan/Version",
+                                         AgencySimpleOperationType::INCREMENT_OP);
+        AgencyWriteTransaction trx(incrementVersion);
+        AgencyCommResult res = _agency.sendTransactionWithFailover(trx, timeout);
+
+        if (res.successful()) {
+          LOG_TOPIC(TRACE, Logger::HEARTBEAT) << "successfully increased plan version in agency";
+        } else {
+          LOG_TOPIC(DEBUG, Logger::HEARTBEAT) << "could not increase version number in agency";
```

Comment:
I suggest to raise this to level "WARN". If this is not working, then DC2DC might not be updating changed collections and we should have something in some logs to diagnose this.